dataRetriever/factory/shard/resolversContainerFactory.go

```diff
@@ -183,12 +183,15 @@ func (rcf *resolversContainerFactory) generateTxResolvers(topic string, unit dat
 func (rcf *resolversContainerFactory) createTxResolver(topic string, excludedTopic string, unit dataRetriever.UnitType) (dataRetriever.Resolver, error) {
 	txStorer := rcf.store.GetStorer(unit)
 
+	//TODO instantiate topic sender resolver with the shard IDs for which this resolver is supposed to serve the data
+	// this will improve the serving of transactions as the searching will be done only on 2 sharded data units
 	resolverSender, err := topicResolverSender.NewTopicResolverSender(
 		rcf.messenger,
 		topic,
 		excludedTopic,
 		rcf.marshalizer,
 		rcf.intRandomizer,
+		uint32(0),
```

Comment:
if it is zero it is considered as every shard ?

Reply:
No, it is not taken into account by the tx resolver. It is a dummy value. There is a TODO to improve the performance on the tx resolver that will be done in a future task